logger.go

```diff
@@ -0,0 +1,13 @@
+package workflow
+
+import "context"
+
+type Logger interface {
+	// Debug will be used by workflow for debug logs when in debug mode.
+	Debug(ctx context.Context, msg string, meta map[string]string)
```

Comment:
Was that this would only log if DebugMode is enabled. That way it works similiar to "log/slog"'s log level where the application only writes calls to logs that match the logger's log level. For example:

 "log/slog" doesnt log Debug logs unless you specify LevelDebug.
 
 For now I have chosen to keep the code more readable by checking w.DebugMode and then calling logger.Debug if its enabled as opposed to the implecit and less readable approach.